src/Icicle/Pipeline.hs

```diff
@@ -296,10 +297,10 @@ flattenAvalanche av
  = join
  . second snd
  . first CompileErrorFlatten
- $ Fresh.runFreshT go (freshNamer "flat")
+ $!! Fresh.runFreshT go (freshNamer "flat")
  where
   go = do s' <- AS.flatten () (AP.statements av)
-          return $ checkAvalanche (av { AP.statements = s' })
+          return $ checkAvalanche (av { AP.statements = force s' })
```

Comment:
@amosr so this exact amount of strictness proved to be the sweet spot (time ~14m, space < 8GB). If I make things more strict, ice eats up so much memory it gets killed by the OS. I'll play around with more strictness in another PR.